app/controllers/concerns/back_links.rb

```diff
@@ -0,0 +1,46 @@
+# Extract and collect helper methods relating to backlinks
+#
+module BackLinks
+  extend ActiveSupport::Concern
+
+  def return_to_after_edit(default:)
+    if redirect_back_to_application_review_page?
+      { back_path: candidate_interface_application_review_path, params: redirect_back_to_application_review_page_params }
+    else
+      { back_path: default, params: {} }
+    end
+  end
+
+  def redirect_back_to_application_review_page_params
+    { 'return-to' => 'application-review' }
+  end
+
+  def redirect_back_to_application_review_page?
+    params['return-to'] == 'application-review' || params[:return_to] == 'application-review'
+  end
+
+  # Method to determine the path to the candidates current dashboard based on
+  # contextual information. For continuous applciations, the dahsboard path
+  def application_form_path
+    # `current_application` is a `helper_method` defined in CandidateInterfaceController
+    # It's not available in view specs
+    return '' unless defined?(current_application)
+
+    if current_application.continuous_applications?
+      if request.path.match?(/withdraw/)
+        candidate_interface_continuous_applications_choices_path
+      else
+        candidate_interface_continuous_applications_details_path
+      end
+    elsif !current_application.submitted?
+      candidate_interface_application_form_path
+    elsif current_application.submitted?
```

Comment:
An `else` here is probably enough.

Reply:
Maybe split the nested if statement into it's own method too?

Reply:
@avinhurry can you suggest what the method should look like?

Reply:
I'd do something like
```ruby
def application_form_path
  # `current_application` is a `helper_method` defined in CandidateInterfaceController
  # It's not available in view specs
  return '' unless defined?(current_application)

  if current_application.continuous_applications?
    continuous_application_form_path
 else
    old_cycle_application_form_path
 end
end

def continuous_application_form_path
  if request.path.match?(/withdraw/)
    candidate_interface_continuous_applications_choices_path
  else
    candidate_interface_continuous_applications_details_path
  end
end

def old_cycle_application_form_path
  if current_application.submitted?
    candidate_interface_application_review_submitted_path
  else
    candidate_interface_application_form_path
  end
end
```

I put "old cycle" but can be other name. This method won't be needed after 3rd October anyway. But it will be easier to delete later.